kernel/sched.c

```diff
@@ -720,19 +723,43 @@ void z_unpend_thread(struct k_thread *thread)
 bool z_thread_prio_set(struct k_thread *thread, int prio)
 {
 	bool need_sched = 0;
+	int old_prio = thread->base.prio;
 
 	K_SPINLOCK(&_sched_spinlock) {
 		need_sched = z_is_thread_ready(thread);
 
 		if (need_sched) {
-			/* Don't requeue on SMP if it's the running thread */
 			if (!IS_ENABLED(CONFIG_SMP) || z_is_thread_queued(thread)) {
 				dequeue_thread(thread);
 				thread->base.prio = prio;
 				queue_thread(thread);
+
+				if (IS_ENABLED(CONFIG_SMP) && (old_prio > prio)) {
+					/*
+					 * IPIs are needed when raising the
+					 * priority of a queued thread.
+					 */
+
+					flag_ipi();
```

Comment:
Style nit: Can

you

use

fewer

lines?

Seriously it's a one-line thing, I really don't think you need a comment, and certainly not extra whitespace.  Also no need to test SMP as flag_ipi() is already a stub if that's not active.

```c
if (old_prio > prio) {
     flag_ipi();
}
```

Reply:
Man, this comment made me immediately laugh. It still brings a chuckle every time I read it. :) I  can amend this.